s3/aide.go

```diff
@@ -94,3 +94,34 @@ func (svc *Service) SetACL(v *string) {
 func (svc *Service) SetSSE(v *string) {
 	svc.sse = v
 }
+
+// ListObjectsKeysV2Pages will list the bucket keys page-wise
+func (svc *Service) ListObjectsKeysV2Pages(params *s3.ListObjectsV2Input) ([]string, bool, error) {
+
+	var keys []string
+	var islastPage bool
+	listObjectsOutputFn := func(page *s3.ListObjectsV2Output, lastPage bool) bool {
+		if page.Contents != nil {
+			for _, obj := range page.Contents {
+				if obj.Key == nil {
+					continue
+				}
+				keys = append(keys, *obj.Key)
+			}
+			islastPage = lastPage
+		}
+		return false
+	}
+
+	err := svc.svc.ListObjectsV2Pages(params, listObjectsOutputFn)
+
+	if err != nil {
+		return nil, islastPage, err
+	}
```

Comment:
This can be replaced by the shorthand:

```go
if err := svc.svc.ListObjectsV2Pages(params, listObjectsOutputFn); err != nil {
	return nil, isLastPage, err
}
```

Reply:
Done.